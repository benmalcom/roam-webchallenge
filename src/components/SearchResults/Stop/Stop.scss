*{
  box-sizing: border-box;
}
.stop-item{
  margin: 5px auto;
  height: 110px;
  min-height: 110px;
  max-height: 110px;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  background-color: #fff;
  .name{
    font-size: 17px;
  }
  .marker-image{
    height: 100%;
    width: 100%;
    border-top-left-radius: 8px;
    border-bottom-left-radius: 8px;
  }
  .stop-item-details{
    height: 100%;
    max-height: 100%;
    overflow-y: auto;
    padding: 10px 5px;
  }
}

.shadow{
  box-shadow: 0 0 1px 0 hsla(0,0%,0%,0.2);
}

/*

##Device = Most of the Smart phones Mobiles (Portrait)
##Screen = B/w 320px to 479px
*/
@media (min-width: 320px) and (max-width: 480px){

}
/*
  ##Device = Low Resolution Tablets, Mobiles (Landscape)
  ##Screen = B/w 481px to 767px
*/
@media (min-width: 481px) and (max-width: 767px) {

}

/*
##Device = Tablets, Ipads (landscape)
##Screen = B/w 768px to 1024px
*/
@media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape){

}

/*
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/
@media (min-width: 768px) and (max-width: 1024px) {

}

/*
  ##Device = Laptops, Desktops
  ##Screen = B/w 1025px to 1280px
*/
@media (min-width: 1025px) and (max-width: 1280px) {

}

/*
  ##Device = Desktops
  ##Screen = 1281px to higher resolution desktops
*/
@media (min-width: 1281px) {

}
