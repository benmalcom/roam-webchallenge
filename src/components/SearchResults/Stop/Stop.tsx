import React from 'react';
const locationMarker = require('./marker.jpg');
import './Stop.scss';

export interface StopProps {
	id?: string;
	name: string;
	desc: string;
	distance: number;
}

export default ({name, desc, distance}: StopProps) => {
	return (<div className="col-lg-4 col-md-4 col-sm-6 col-12">
		<div className="row stop-item shadow">
			<div className="col-md-4 col-4 full-height full-width p-0">
				<img src={locationMarker} className="marker-image"/>
			</div>
			<div className="col-md-8 col-8 pt-30 stop-item-details">
				<div className="mr-auto text-left">
					<h6 className="mb-1 mt-1">
						<strong className="name">{name}</strong>
					</h6>
					<h6 className="text-sm text-muted">{desc}</h6>
					<label className="badge badge-info text-xs">
						<i className="fa fa-car"/> {distance} meters away
					</label>

				</div>
			</div>
		</div>
	</div>);
};
