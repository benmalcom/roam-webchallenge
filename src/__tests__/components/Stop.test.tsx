import React from 'react';
import Stop from '../../components/SearchResults/Stop/Stop';
import { mount } from 'enzyme';
describe('<Stop />', () => {
	let props;
	let component;

	beforeEach(() => {
		props = {
			name: 'A new location',
			desc: 'This is the description',
			distance: 112,
		};
	});
	test('Component renders properly with props',  () => {
		component = mount(<Stop {...props}/>);
		expect(component).toMatchSnapshot();
		expect(component.prop('name')).toEqual(props.name);
		expect(component.prop('desc')).toEqual(props.desc);
		expect(component.prop('distance')).toEqual(props.distance);
	});
});
