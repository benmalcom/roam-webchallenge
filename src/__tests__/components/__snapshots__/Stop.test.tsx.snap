// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stop /> Component renders properly with props 1`] = `
<Component
  desc="This is the description"
  distance={112}
  name="A new location"
>
  <div
    className="col-lg-4 col-md-4 col-sm-6 col-12"
  >
    <div
      className="row stop-item shadow"
    >
      <div
        className="col-md-4 col-4 full-height full-width p-0"
      >
        <img
          className="marker-image"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
      <div
        className="col-md-8 col-8 pt-30 stop-item-details"
      >
        <div
          className="mr-auto text-left"
        >
          <h6
            className="mb-1 mt-1"
          >
            <strong
              className="name"
            >
              A new location
            </strong>
          </h6>
          <h6
            className="text-sm text-muted"
          >
            This is the description
          </h6>
          <label
            className="badge badge-info text-xs"
          >
            <i
              className="fa fa-car"
            />
             
            112
             meters away
          </label>
        </div>
      </div>
    </div>
  </div>
</Component>
`;
